docs/configuration.md

```diff
@@ -0,0 +1,82 @@
+
+As per the requirement the following fields can be changed.
+
+## targetNameSpaces
+
+Multiple namespaces can be specified to deploy partner pods for testing through `targetNameSpaces` in the config file.
+
+``` { .yaml .annotate }
+targetNameSpaces:
+  - name: firstnamespace
+  - name: secondnamespace
+```
+## targetPodLabels
+The goal of this section is to specify the labels to be used to identify the CNF resources under test.
+
+!!! note "Highly recommended"
+
+    The labels should be defined in pod definition rather than added after pod is created, as labels added later on will be lost in case the pod gets rescheduled. In case of pods defined as part of a deployment, it's best to use the same label as the one defined in the `spec.selector.matchLabels` section of the deployment yaml. The prefix field can be used to avoid naming collision with other labels.
+
+``` { .yaml .annotate }
+targetPodLabels:
+  - prefix: test-network-function.com
+    name: generic
+    value: target
+```
+
+The corresponding pod label used to match pods is:
+``` { .yaml .annotate }
+test-network-function.com/generic: target
+```
+
+Once the pods are found, all of their containers are also added to the target container list. A target deployment list will also be created with all the deployments which the test pods belong to.
+
+## targetCrds
+In order to autodiscover the CRDs to be tested, an array of search filters can be set under the "targetCrdFilters" label. The autodiscovery mechanism will iterate through all the filters to look for all the CRDs that match it. Currently, filters only work by name suffix.
+
+``` { .yaml .annotate }
+targetCrdFilters:
+ - nameSuffix: "group1.tnf.com"
+ - nameSuffix: "anydomain.com"
+```
+
+The autodiscovery mechanism will create a list of all CRD names in the cluster whose names have the suffix `group1.tnf.com` or `anydomain.com`, e.g. `crd1.group1.tnf.com` or `mycrd.mygroup.anydomain.com`.
+
+## testTarget
+### podsUnderTest / containersUnderTest
+The autodiscovery mechanism will attempt to identify the default network device and all the IP addresses of the pods it needs for network connectivity tests, though that information can be explicitly set using annotations if needed.
+
+
+#### Pod IPs
+
+* The `k8s.v1.cni.cncf.io/networks-status` annotation is checked and all IPs from it are used. This annotation is automatically managed in OpenShift but may not be present in K8s.
+* If it is not present, then only known IPs associated with the pod are used (the pod `.status.ips` field).
+
+#### Network Interfaces
+
+* The annotation `test-network-function.com/defaultnetworkinterface` is the highest priority, and must contain a JSON-encoded string of the primary network interface for the pod. This must be explicitly set if needed. Examples are provided in [cnf-certification-test-partner](https://github.com/test-network-function/cnf-certification-test-partner).
```

Comment:
I'm not sure whether this is still applying. I haven't found any reference to this annotation in the code. @edcdavid can you confirm this, please?

Reply:
I found a reference to it here: https://github.com/test-network-function/cnf-certification-test-partner/blob/main/test-target/local-pod-under-test.yaml#L22

But it has an `#optional` comment on it?